lib/rubocop/ast/node.rb

```diff
@@ -508,6 +508,21 @@ def guard_clause?
          (block (send (const nil? {:Class :Module}) :new ...) ...)}
       PATTERN
 
+      def_node_matcher :struct_constructor?, <<~PATTERN
+        (block (send (const {nil? (cbase)} :Struct) :new ...) ...)
+      PATTERN
+
+      def_node_matcher :class_definition?, <<~PATTERN
+        {(class (const _ _) ...)
+         (casgn nil? _ (block (send (const {nil? (cbase)} {:Struct :Class}) :new ...) ...))
+         (sclass {(self) (send nil? _)} ...)}
```

Comment:
Same here... The only point in being precise is to exclude something. There are no `sclass` to exclude, and by being more precise you create a pattern that will be slower (for nothing) or wrong (here you are excluding `class << obj` by mistake)